pkg/deployment/collector_test.go

```diff
@@ -486,3 +486,28 @@ func TestAutoscalersDisabledByExplicitOption(t *testing.T) {
 	// verify
 	assert.Len(t, a, 0)
 }
+
+func TestCollectorOrderOfArgumentsOpenshiftTLS(t *testing.T) {
+	viper.Set("platform", v1.FlagPlatformOpenShift)
+	jaeger := v1.NewJaeger(types.NamespacedName{Name: "my-instance"})
+	jaeger.Spec.Collector.Options = v1.NewOptions(map[string]interface{}{
+		"b-option": "b-value",
+		"a-option": "a-value",
+		"c-option": "c-value",
+	})
+
+	a := NewCollector(jaeger)
+	dep := a.Get()
+
+	assert.Len(t, dep.Spec.Template.Spec.Containers, 1)
+	assert.Len(t, dep.Spec.Template.Spec.Containers[0].Args, 7)
+	assert.True(t, strings.HasPrefix(dep.Spec.Template.Spec.Containers[0].Args[0], "--a-option"))
```

Comment:
I think we have other tests with better assertions around this. Take a look at the `util` package, I think we have a utility to get arguments from the options map.